dockstore-webservice/src/main/java/io/dockstore/webservice/core/FileFormat.java

```diff
@@ -42,7 +43,12 @@
 @ApiModel(value = "FileFormat", description = "This describes an input or output file format that is associated with an entry in the dockstore")
 @Entity
 @Table(name = "fileformat")
-@NamedQuery(name = "io.dockstore.webservice.core.FileFormat.findByFileFormatValue", query = "SELECT l FROM FileFormat l WHERE l.value = :fileformatValue")
+@NamedQueries({
+        @NamedQuery(name = "io.dockstore.webservice.core.FileFormat.findByFileFormatValue", query = "SELECT l FROM FileFormat l WHERE l.value = :fileformatValue"),
+        @NamedQuery(name = "io.dockstore.webservice.core.FileFormat.findInputFileFormatsInEntry", query = "Select inputFileFormats FROM Version version INNER JOIN version.inputFileFormats as inputFileFormats WHERE version.parent.id = :entryId"),
```

Comment:
Will this potentially return duplicates? And then the duplicates are removed because `Entry.inputFileFormats` is a `TreeSet`? If yes, then you should have the DB do the deduplication work and do `select distinct(inputFileFormats)...`, assuming that works. And then `Entry.inputFileFormats` could just be a List.

Update: looking at code further down below, maybe you don't want to make it a List, but I it would still be better to  have the DB do the de-duplication.

Reply:
Oh think I will remove this, I didn't end up using it.